testng-core/src/main/resources/testng-1.0.dtd

```diff
@@ -73,6 +75,7 @@ Cedric Beust & Alexandru Popescu
     time-out CDATA #IMPLIED
     skipfailedinvocationcounts (true | false) "false"
     data-provider-thread-count CDATA "10"
+    share-thread-pool-for-data-providers (true | false) "false"
```

Comment:
Sure. Can you please guide me on the following?

1. When do we release a new dtd version? That way we have a process that we can follow whenever there are dtd changes. 
2. Assuming that we have the process defined, how do we go about deploying the new dtd.
3. Since a user can still have their suite xml dtd point to the newer one, how would the new dtd publishing help?

Reply:
DTD (or XSD) is a contract like an interface.
If you want to modify it, you should release a new version and be able to manage all versions in the XML reader.

Reply:
@juherr - Sure. I understand the intent of the DTD/XSD. But what purpose does this solve for TestNG is what I am trying to understand. Our users are mostly going to be on the same XSD (asking them to go update all the suite files with pointing it to a newer version is kind of like a sub-optimal user experience for me). The XSD exists only because we would like to validate our xml schema and it really doesn't serve any functional purpose from the user's perspective. So why would they want to update. So eventually what benefit does TestNG have by doing this? On the other hand, if we just go and update the same XSD with newer attributes, a user would just need to update to the newer TestNG version that is aware of the newer attributes and this will just work for a user. This is what my concern is.